pyuvdata/utils.py

```diff
@@ -2071,3 +2072,254 @@ def apply_uvflag(
 
     if not inplace:
         return uvd
+
+
+def parse_ants(uv, ant_str, print_toggle=False, x_orientation=None):
+    """
+    Get antpair and polarization from parsing an aipy-style ant string.
+
+    Used to support the the select function.
+    Generates two lists of antenna pair tuples and polarization indices based
+    on parsing of the string ant_str.  If no valid polarizations (pseudo-Stokes
+    params, or combinations of [lr] or [xy]) or antenna numbers are found in
+    ant_str, ant_pairs_nums and polarizations are returned as None.
+
+    Parameters
+    ----------
+    uv : UVBase Object
+        A UVBased object that supports the following functions and parameters:
+        - get_ants
+        - get_antpairs
+        - get_pols
+        These are used to construct the baseline ant_pair_nums
+        and polarizations returned.
+    ant_str : str
+        String containing antenna information to parse. Can be 'all',
+        'auto', 'cross', or combinations of antenna numbers and polarization
+        indicators 'l' and 'r' or 'x' and 'y'.  Minus signs can also be used
+        in front of an antenna number or baseline to exclude it from being
+        output in ant_pairs_nums. If ant_str has a minus sign as the first
+        character, 'all,' will be appended to the beginning of the string.
+        See the tutorial for examples of valid strings and their behavior.
+    print_toggle : bool
+        Boolean for printing parsed baselines for a visual user check.
+    x_orientation : str, optional
+        Orientation of the physical dipole corresponding to what is
+        labelled as the x polarization ("east" or "north") to allow for
+        converting from E/N strings. See corresonding parameter on UVData
+        for more details.
```

Comment:
my read of the code now is that if you don't pass this explicitly it won't be set, even if it is set on the object. Is that what we want?

Reply:
Yeah that's the way it is written currently. Could have it pick up the value from the object if it exists and input was None, but use the optional parameter to override if given?

Reply:
I'd vote for that. Then you'll need to make it clear in the docstring that it overrides the value on the object.

Reply:
@mkolopanis do you want to make this change or just leave it as is?

Reply:
Oh yes I do want to make this change. Thanks for the ping.

Reply:
I think you mean `override` not `overwrite`. It won't actually change the value on the object will it?

Reply:
good catch definitely a typo.